library/Reporting/Report.php

```diff
@@ -30,98 +30,39 @@ class Report
+    public static function fromModel(Model\Report $reportModel): Report
     {
         $report = new static();
 
-        $db = $report->getDb();
-
-        $select = (new Sql\Select())
-            ->from('report')
-            ->columns('*')
-            ->where(['id = ?' => $id]);
+        $report->id = $reportModel->id;
+        $report->name = $reportModel->name;
+        $report->author = $reportModel->author;
+        $report->timeframe = Timeframe::fromModel($reportModel->timeframe);
+        $report->template = Template::fromModel($reportModel->template);
 
-        $row = $db->select($select)->fetch();
-
-        if ($row === false) {
-            throw new Exception('Report not found');
-        }
-
-        $report
-            ->setId($row->id)
-            ->setName($row->name)
-            ->setAuthor($row->author)
-            ->setTimeframe(Timeframe::fromDb($row->timeframe_id))
-            ->setTemplate(Template::fromDb($row->template_id));
-
-        $select = (new Sql\Select())
-            ->from('reportlet')
-            ->columns('*')
-            ->where(['report_id = ?' => $id]);
-
-        $row = $db->select($select)->fetch();
-
-        if ($row === false) {
+        if ($reportModel->reportlet === null) {
             throw new Exception('No reportlets configured.');
         }
 
-        $reportlet = new Reportlet();
-
-        $reportlet
-            ->setId($row->id)
-            ->setClass($row->class);
+        $reportModel->reportlet->report_name = $reportModel->name;
+        $reportModel->reportlet->report_id = $reportModel->id;
+        $reportlet = Reportlet::fromModel($reportModel->reportlet);
 
-        $select = (new Sql\Select())
-            ->from('config')
-            ->columns('*')
-            ->where(['reportlet_id = ?' => $row->id]);
+        $report->reportlets = [$reportlet];
 
-        $rows = $db->select($select)->fetchAll();
-
-        $config = [
-            'name'  => $report->getName(),
-            'id'    => $report->getId()
-        ];
-
-        foreach ($rows as $row) {
-            $config[$row->name] = $row->value;
-        }
-
-        $reportlet->setConfig($config);
-
-        $report->setReportlets([$reportlet]);
-
-        $select = (new Sql\Select())
-            ->from('schedule')
-            ->columns('*')
-            ->where(['report_id = ?' => $id]);
-
-        $row = $db->select($select)->fetch();
-
-        if ($row !== false) {
-            $schedule = new Schedule();
-
-            $schedule
-                ->setId($row->id)
-                ->setStart((new \DateTime())->setTimestamp((int) $row->start / 1000))
-                ->setFrequency($row->frequency)
-                ->setAction($row->action)
-                ->setConfig(json_decode($row->config, true));
-
-            $report->setSchedule($schedule);
+        if ($reportModel->schedule->id) {
+            $report->schedule = Schedule::fromModel($reportModel->schedule);
```

Comment:
If you wouldn't include `schedule` within the `ReportController`, you can do this as follows.
```suggestion
        $schedule = $reportModel->schedule->first();
        if ($schedule !== null) {
            $report->schedule = Schedule::fromModel($schedule);
        }
```